sdk/azidentity/client_certificate_credential.go

```diff
@@ -56,35 +53,18 @@ type ClientCertificateCredential struct {
 // NewClientCertificateCredential creates an instance of ClientCertificateCredential with the details needed to authenticate against Azure Active Directory with the specified certificate.
 // tenantID: The Azure Active Directory tenant (directory) ID of the service principal.
 // clientID: The client (application) ID of the service principal.
-// certificatePath: The path to the client certificate used to authenticate the client.  Supported formats are PEM and PFX.
+// certificateData: The bytes of a certificate in PEM or PKCS12 format, including the private key.
```

Comment:
Presumably this is just the bytes from a call to `ioutil.ReadAll()` or something similar.  Assuming that's the case, it would be good to add an example demonstrating this.

Reply:
Another option would be to take an `io.Reader` instead which might cut down on the amount of code the caller has to write.

Reply:
I picked the slice because it's the lowest common denominator and we need all the data in memory to deserialize it anyway. I suppose the question is whether a caller is more likely to have `[]byte` or `io.Reader`. Do you think we create more struggle overall by requiring the slice?

Reply:
I don't think it creates more struggle (as you mentioned, it's a trivial amount of code), I'm just wondering if an `io.Reader` will provide more flexibility in the future.

The key vault example is an interesting one.  I don't know what the return type looks like for that when retrieving a cert, but if it's just binary data it would be returned as an `io.ReadCloser`.  In that case we could simply pass the result to this constructor without requiring the caller to read it themselves.

Do you know what other languages accept here?

Reply:
The value from Key Vault would be a plain `string` (i.e. not binary). .NET and Python expose it as `string`.

Every SDK's `ClientCertificateCredential` has a constructor taking a path (but I think consistency in that is not important for us because we only get one "good" constructor). Everyone but JS takes an alternative as well: for .NET it's `X509Certificate2`; for Java, a byte stream; for Python, non-streaming bytes.

Reply:
Interesting, surprises me it's a string.  Is the underlying data base-64 encoded?

Reply:
Only for PKCS12 certs. So to consume the value from Key Vault directly, we would have to consider that cert data not in PEM format might be base64 encoded PKCS12 content. For Python, I left the decoding up to applications and added a sample showing how to do it.